controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -189,6 +189,23 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
+	profileName := r.StorageClassRequest.Spec.StorageProfile
+	if profileName == "" {
+		profileName = r.storageCluster.Spec.DefaultStorageProfile
+	}
+
+	// Fetch StorageProfile by name in the StorageCluster's namespace
+	storageProfile := v1.StorageProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      profileName,
+			Namespace: r.storageCluster.Namespace,
+		},
+	}
+
+	if err := r.get(&storageProfile); err != nil {
+		return reconcile.Result{}, fmt.Errorf("no storage profile CR found for storage profile %s", profileName)
+	}
+
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
```

Comment:
Why are we running this code here and not in the dedicated `r.reconcileCephBlockPool` below?
Same question for the handling of the SVG.

Reply:
A good question. I would say it depends on the amount of change needed and the complexity (if any) of the changes.
I will leave it to you to decide if we want it here or in a following PR

Reply:
I'll make the change in a separate commit and see what it looks like.

Reply:
@jarrpa Can we get to a resolution on this thread? Should we wait for a second commit? or are you going to open a github issue and link it here?

Reply:
We do not need to wait for a second commit, no. I poked and it and determined it's worth having its own separate discussion, hopefully a brief one. I'll submit a follow-up draft PR for it to show what I found.